MAINTAINING.md

```diff
@@ -117,6 +117,20 @@ complete that no input from the original developer or maintainer is required.
 -   Be friendly. Respect the original author, bearing in mind that their coding
     style or their design may be just as valid as the way you would have done
     it. And of course, always follow the [Code of Conduct].
+-   If a contributor has opened a PR, the reviewer should attempt to
+    help the author of the contribution to get it to its best shape and
+    according to the RIOT Standard™. If there is disagreement, it’s important
+    to understand the reasons behind and always give technical arguments,
+    pros and cons or snippets.
+-   If a PR is stalled or there’s the need to open a PR with
+    the intention to be merged, it should be discussed with the original authors
+    before proceeding.
+-   If the contribution is stalled or is a challenge to explain how to improve a
+    demonstrator PR can be opened to better clarify.
+-   If the contribution remains below the quality for RIOT then a request should
+    be issued to the contributor asking if a competing PR can be opened.
+    If there is no response after a reasonable amount of time then the competing
+    PR can be open with a reference to the initial contributing PR.
```

Comment:
What would we do if someone responds with "no"? Would we really care?

We often have clean up PRs that improve on existing code in RIOT's code base. So far, we have decided based solely on the technical arguments. If a PR indeed improves objectively RIOT's code base and is up to standards, would we ask the original author of the code for permission? (Obviously often the original authors is requested to review, because that author often has the best technical background for doing a review. But we wouldn't stall a PR if there is a rough consensus that it improves RIOT, even if the original author is sees this differently.)

I think we should have the same standard for PRs. If one is technically better and there is no value in having two solutions, IMO we should just merge the better one.